main.tf

```diff
@@ -227,6 +242,33 @@ module "wandb" {
         ]
       }
 
+      ## To support otel azure monitor sql and redis metrics need operator-wandb chart minimum version 0.14.0 
+      otel = {
+        daemonset =  {
+          pod = var.azuremonitor ? { labels = { "azure.workload.identity/use" = "true" } } : {}
+          presets = var.azuremonitor ? {
+            receiver = {
+              azuremonitor = {
+                subscription_id = data.azurerm_subscription.current.subscription_id
+                resource_groups = var.namespace
+              }
+            }
+          } : {}
+          config = var.azuremonitor ? {
+            service = {
+              pipelines = {
+                metrics = {
+                  receivers = ["hostmetrics", "k8s_cluster", "kubeletstats", "azuremonitor"]
+                }
+              }
+            }
+          } : {}
+          serviceAccount = var.azuremonitor ? { 
+            annotations = { "azure.workload.identity/client-id" = module.identity.identity.client_id }
+            } : {}
+        } 
```

Comment:
Why are we making it a preset? Lets just set it here directly? 

It might also be more readbale to wrap daemonset = `var.azuremonitor ? ...` we can always break it out after